File: CS/CustomGridWinApp1/CustomGridControl/MyOptionsView.cs
using System.ComponentModel;
using DevExpress.Utils;
using DevExpress.Utils.Controls;
using DevExpress.Utils.Serializing;
using DevExpress.XtraGrid.Views.Grid;

namespace CustomGridControl
{
	public class MyOptionsView : GridOptionsView
	{
		private VertAlignment detailButtonsVAlignment;

		public MyOptionsView()
		{
			this.detailButtonsVAlignment = VertAlignment.Default;
		}

		public override void Assign(BaseOptions options)
		{
			this.BeginUpdate();
			try
			{
				base.Assign(options);

				MyOptionsView optView = options as MyOptionsView;
				if ( optView == null )
					return;

				this.detailButtonsVAlignment = optView.DetailButtonsVAlignment;
			} finally
			{
				this.EndUpdate();
			}
		}

		[Description("Gets or sets the master-detail expand buttons vertical alignment."), DefaultValue(VertAlignment.Default), XtraSerializableProperty()]
		public VertAlignment DetailButtonsVAlignment
		{
			get { return detailButtonsVAlignment; }
			set
			{
				if ( detailButtonsVAlignment == value )
					return;

				VertAlignment prevValue = DetailButtonsVAlignment;
				detailButtonsVAlignment = value;
				OnChanged(new BaseOptionChangedEventArgs("DetailButtonVAlignment", prevValue, DetailButtonsVAlignment));
			}
		}
	}
}

File: VB/CustomGridWinApp1/CustomGridControl/MyOptionsView.vb
Imports System.ComponentModel
Imports DevExpress.Utils
Imports DevExpress.Utils.Controls
Imports DevExpress.Utils.Serializing
Imports DevExpress.XtraGrid.Views.Grid

Namespace CustomGridControl
	Public Class MyOptionsView
		Inherits GridOptionsView
		Private detailButtonsVAlignment_Renamed As VertAlignment

		Public Sub New()
			Me.detailButtonsVAlignment_Renamed = VertAlignment.Default
		End Sub

		Public Overrides Sub Assign(ByVal options As BaseOptions)
			Me.BeginUpdate()
			Try
				MyBase.Assign(options)

				Dim optView As MyOptionsView = TryCast(options, MyOptionsView)
				If optView Is Nothing Then
					Return
				End If

				Me.detailButtonsVAlignment_Renamed = optView.DetailButtonsVAlignment
			Finally
				Me.EndUpdate()
			End Try
		End Sub

		<Description("Gets or sets the master-detail expand buttons vertical alignment."), DefaultValue(VertAlignment.Default), XtraSerializableProperty()> _
		Public Property DetailButtonsVAlignment() As VertAlignment
			Get
				Return detailButtonsVAlignment_Renamed
			End Get
			Set(ByVal value As VertAlignment)
				If detailButtonsVAlignment_Renamed = value Then
					Return
				End If

				Dim prevValue As VertAlignment = DetailButtonsVAlignment
				detailButtonsVAlignment_Renamed = value
				OnChanged(New BaseOptionChangedEventArgs("DetailButtonVAlignment", prevValue, DetailButtonsVAlignment))
			End Set
		End Property
	End Class
End Namespace
